### PR TITLE
Feature/4131 verifycode screen

### PR DESCRIPTION
**Pull request by RaghavendranC-Eaton**
Fixes # Verify code scree page updated and completed unit test cases.

#### Changes proposed in this Pull Request:

-
-
-

#### Screenshots / Screen Recording (if applicable)

- 
Screenshot 2023-06-28 at 12 48 42 PM

Screenshot 2023-06-28 at 12 48 57 PM

#### To Test:

- yarn start:example2
- Click the Registration workflow and see the verify code screen page 

#### Any specific feedback you are looking for?

-

### CODE REVIEW

File: login-workflow/example2/src/screens/new-architecture-test-screens/RegistrationWorkflowScreen.tsx [hunk 1]
```diff
@@ -27,6 +28,7 @@ export const RegistrationWorkflowScreen = (): JSX.Element => {
             <RegistrationWorkflow initialScreenIndex={0}>
                 <EulaScreen />
                 <CreatePasswordScreen />
+                <VerifyCodeScreen />
```
Comment (surajeaton) [inline login-workflow/example2/src/screens/new-architecture-test-screens/RegistrationWorkflowScreen.tsx:31]: This screen should be before  <CreatePasswordScreen />

File: login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+export const VerifyCodeScreen: React.FC<VerifyCodeScreenProps> = (props) => {
+    const { t } = useLanguageLocale();
+    const regWorkflow = useRegistrationWorkflowContext();
+    const { actions } = useRegistrationContext();
+    const { nextScreen, previousScreen, screenData } = regWorkflow;
+
+    const {
+        codeValidator = (code: string): boolean | string =>
+            code?.length > 0 ? true : t('bluiRegistration:SELF_REGISTRATION.VERIFY_EMAIL.CODE_VALIDATOR_ERROR'),
+        onResend = (): void => {
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreen.tsx:29]: We should not console, instead we have to call an action.

File: login-workflow/src/new-architecture/contexts/RegistrationContext/dictionaries/english.ts [hunk 3]
```diff
@@ -35,9 +35,10 @@ const resources: RegistrationLanguageFile = {
             VERIFY_EMAIL: {
                 MESSAGE:
                     'A verification code has been sent to the email address you provided. Click the link or enter the code below to continue. This code is valid for 30 minutes.',
-                RESEND: 'Resend Verification Email',
+                RESEND: 'Send Again',
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/contexts/RegistrationContext/dictionaries/english.ts:38]: No need to update this key. There is existing key shared directory `ACTIONS.RESEND`

File: login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreen.test.tsx [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+import { VerifyCodeScreen } from './VerifyCodeScreen';
+import { VerifyCodeScreenProps } from './types';
+import {
+    RegistrationContextProvider,
+    RegistrationContextProviderProps,
+    i18nRegistrationInstance,
+} from '../../contexts';
+import { RegistrationWorkflow } from '../../components';
+
+const defaultProps: RegistrationContextProviderProps = {
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreen.test.tsx:13]: You can use the existing defaultProps:
`import { defaultProps as registrationContextProviderProps } from '../../contexts/RegistrationContext/RegistrationContextProvider.test';`

**Comment by RaghavendranC-Eaton** [comment]
@surajeaton I have fixed the PR comments. Pls review.

File: login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreenBase.tsx
Comment (surajeaton) [inline login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreenBase.tsx:1]: There should be no default defined in the based screen.
On line 27: 
codeValidator prop has given the default value, can you please remove it.

File: login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+                screenId: 'VerifyCode',
+                values: { code: verifyCode },
+            });
+            setIsLoading(false);
+        } catch {
+            console.error('Error fetching validation code!');
+        }
+    }, [verifyCode, nextScreen, actions]);
+
+    const onPrevious = useCallback(() => {
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreen.tsx:53]: There is no need of `useCallback` in onPrevious

File: login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreen.test.tsx [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+                </RegistrationWorkflow>
+            </RegistrationContextProvider>
+        );
+
+        const verifyCodeInput = getByLabelText('Verify Code');
+        fireEvent.change(verifyCodeInput, { target: { value: '1234' } });
+        fireEvent.blur(verifyCodeInput);
+        expect(screen.getByText(/Next/i)).toBeEnabled();
+        expect(verifyCodeInput).toHaveAttribute('aria-invalid', 'false');
+        //expect(screen.getByText('Code must be at least 3 characters')).toBeInTheDocument();
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreen.test.tsx:105]: Remove this commented assertion.

File: login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+/**
+ * Component that renders a screen that prompts a user to enter the confirmation code
+ * that was sent to the email address that they used to register.
+ *
+ * @param codeValidator used to test the input for valid formatting
+ * @param onResend function to call when the user clicks the 'resend code' button
+ * @param resendInstructions text to display ahead of the resend link/button
+ * @param resendLabel label for the resend link/button
+ * @param initialValue code used to pre-populate the field
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreen.tsx:15]: Add `updateCode ` to params

**Comment by RaghavendranC-Eaton** [comment]
@surajeaton I have fixed the PR comments. Please review

File: login-workflow/example2/src/actions/RegistrationUIActions.tsx [hunk 8]
```diff
@@ -59,6 +59,15 @@ export const ProjectRegistrationUIActions: () => RegistrationUIActions = () => (
         return true;
     },
 
+    validateUserRegistrationRequest: async (validationCode: string, validationEmail?: string): Promise<boolean> => {
```
Comment (ektaghag-eaton) [inline login-workflow/example2/src/actions/RegistrationUIActions.tsx:62]: Add param info like other functions.

for e.g.
`
/**
     * The user has tapped on an email link inviting them to register with the application.
     * The application should validate the code provided by the link.
     *
     * @param validationCode  Registration code provided from the link.
     * @param validationEmail  Email provided from the invitation email link (optional) `?email=addr%40domain.com`.
     *
     * @returns Resolves when the code is valid. True if registration is complete, False if account information is needed.
     *          If the code is not valid a rejection will occur with an error message.
     */
`

**Comment by RaghavendranC-Eaton** [comment]
@ektaghag-eaton I have added the param info into the validateUserRegistrationRequest. Pls review it.

File: login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreen.tsx [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+        WorkflowCardInstructionProps: workflowCardInstructionProps = {
+            instructions: t('bluiRegistration:SELF_REGISTRATION.VERIFY_EMAIL.MESSAGE'),
+        },
+        WorkflowCardActionsProps: workflowCardActionsProps = {
+            showNext: true,
+            nextLabel: t('bluiCommon:ACTIONS.NEXT'),
+            showPrevious: true,
+            previousLabel: t('bluiCommon:ACTIONS.BACK'),
+            canGoPrevious: true,
+            currentStep: 4,
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreen.tsx:77]: @RaghavendranC-Eaton `currentStep` should be 3.

File: login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreen.tsx [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+export const VerifyCodeScreen: React.FC<VerifyCodeScreenProps> = (props) => {
+    const { t } = useLanguageLocale();
+    const regWorkflow = useRegistrationWorkflowContext();
+    const { actions } = useRegistrationContext();
+    const { nextScreen, previousScreen, screenData } = regWorkflow;
+
+    const {
+        codeValidator = (code: string): boolean | string =>
+            code?.length > 0 ? true : t('bluiRegistration:SELF_REGISTRATION.VERIFY_EMAIL.CODE_VALIDATOR_ERROR'),
+        onResend,
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreen.tsx:30]: You need to call an action onResend here as default. It is throwing an error :
Screenshot 2023-06-30 at 12 00 31 PM

File: login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreen.tsx (hunk 9 above)
Comment (RaghavendranC-Eaton) [reply login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreen.tsx]: I have added CurrentStep is 2

File: login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreen.tsx (hunk 10 above)
Comment (RaghavendranC-Eaton) [reply login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreen.tsx]: Fixed. Added call an action to onResend props

**Comment by RaghavendranC-Eaton** [comment]
@surajeaton latest PR comments are resolved. pls review

**Review by surajeaton — APPROVED** [review]

**Pull request merged**